syntax = "proto3";
//package invoke;
option cc_generic_services = true;

message StartActorRequest {
  string actor_name = 1;   // 客户端希望给 actor 取的名字
  bytes args_pickle = 2;   // (args, kwargs)
}

message StartActorResponse {
  bool success = 1;
  string actor_name = 2;
  string error = 3;
}

message InvokeRequest {
  string actor_name = 1;   // 要调用哪个 actor
  string method = 2;       // 方法名
  bytes args_pickle = 3;   // (args, kwargs)
}

message InvokeResponse {
  bool success = 1;
  string ObjectID = 2;
  string SrcParty = 3;
  string error = 4;
}

service Invoker {
  rpc StartActor(StartActorRequest) returns (StartActorResponse);
  rpc Invoke(InvokeRequest) returns (InvokeResponse);
}
